[base-node] Create new orphan chain tip on chain reorg

delta1 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
Updates the `handle_possible_reorg` function to insert a new orphan chain tip from the removed blocks, and a test to check this logic. Some minor typos and comments updated.

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
On reorgs where blocks were removed, the new orphan chain tip was not added to the database. 

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
New unit test that was failing before the fix.

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
* [x] I'm merging against the `development` branch.
* [x] I have squashed my commits into a single commit.

stringhandler commented on base_layer/core/src/chain_storage/blockchain_database.rs:
```diff
@@ -1512,9 +1514,17 @@ fn handle_possible_reorg<T: BlockchainBackend>(
-
     let num_removed_blocks = removed_blocks.len();
 
+    if let Some(block) = removed_blocks.first() {
```
I think this makes more sense just before the last line of `reorganize_chain`. That method is already managing when the db transactions are written, so it's easier to find, and also easier think of `reorganize_chain` as a single DB operation

delta1 replied:
Sounds good, done

stringhandler reviewed (commented):
Just one change, then should be fine to go in

Commit:
[base-node] Create new orphan chain tip on chain reorg

Updates the `reorganize_chain` function to insert a new orphan
chain tip from the removed blocks, and a test to check this logic.
